src/machine/core.cpp

```diff
@@ -0,0 +1,1000 @@
+#include "core.h"
+
+#include "programloader.h"
+#include "utils.h"
+
+using namespace machine;
+
+Core::Core(
+    Registers *regs,
+    FrontendMemory *mem_program,
+    FrontendMemory *mem_data,
+    unsigned int min_cache_row_size,
+    Cop0State *cop0state)
+    : ex_handlers()
+    , hw_breaks() {
+    cycle_c = 0;
+    stall_c = 0;
+    this->regs = regs;
+    this->cop0state = cop0state;
+    this->mem_program = mem_program;
+    this->mem_data = mem_data;
+    this->ex_default_handler = new StopExceptionHandler();
+    this->min_cache_row_size = min_cache_row_size;
+    this->hwr_userlocal = 0xe0000000;
+    if (cop0state != nullptr) {
+        cop0state->setup_core(this);
+    }
+    for (int i = 0; i < EXCAUSE_COUNT; i++) {
+        stop_on_exception[i] = true;
+        step_over_exception[i] = true;
+    }
+    step_over_exception[EXCAUSE_INT] = false;
+}
+
+Core::~Core() {
+    delete ex_default_handler;
+}
+
+void Core::step(bool skip_break) {
+    cycle_c++;
+    emit cycle_c_value(cycle_c);
+    do_step(skip_break);
+}
+
+void Core::reset() {
+    cycle_c = 0;
+    stall_c = 0;
+    do_reset();
+}
+
```

Comment:
Remove all resets and always construct new object. => More simple, less possible states.